pkg/controller/instanceset/in_place_update_util.go

```diff
@@ -261,7 +261,10 @@ func equalBasicInPlaceFields(old, new *corev1.Pod) bool {
 
 func equalResourcesInPlaceFields(old, new *corev1.Pod) bool {
 	if len(old.Spec.Containers) != len(new.Spec.Containers) {
-		return false
```

Comment:
should add comments to explain the following code logic:
These lines're used for scenarios that user have other controllers/webhooks like OpenKruise SidecarSet running. It'll mutate the pod to inject custom containers (e.g. sidecars) in. InstanceSet will tolerate these behaviors and avoid pod recreation if additional containers're injected.